lambda_function.py

```diff
@@ -0,0 +1,48 @@
+import os
+import json
+import urllib3
+import boto3
+from urllib3.exceptions import ConnectTimeoutError
+from boto3 import resource
+from botocore.exceptions import ClientError
+from boto3.dynamodb.conditions import Key, Attr
+
+http = urllib3.PoolManager()
+client = boto3.client('events', region_name="eu-west-2")
+
+
+def get_item_with_id(table, primary_key, primary_key_value):
+    response = table.query(KeyConditionExpression=Key(primary_key).eq(primary_key_value))
+    items = response["Items"]
+
+    if len(items) == 0:
+        raise Exception(f"{len(items)} items have the '{primary_key}' primary key of '{primary_key_value}'")
+    else:
+        return items[0]
+
+
+def add_true_to_ingest_cc_attribute(table, primary_key, sort_key, item_with_id):
+    primary_key_value = item_with_id[primary_key]
+    sort_key_value = item_with_id[sort_key]
+
+    table.update_item(
+        Key={primary_key: primary_key_value, sort_key: sort_key_value},
+        UpdateExpression="SET ingested_CC = :ingestedCCValue",
+        ExpressionAttributeValues={":ingestedCCValue": "true"},
+        ConditionExpression=f"attribute_exists({primary_key})"
+    )
+
+
+def get_message_from_json_event(event):
+    return json.loads(event['Records'][0]['Sns']['Message'])
+
+
+def lambda_handler(event, context):
+    message_json_as_dict = get_message_from_json_event(event)
+    primary_key_value = message_json_as_dict["tableItemIdentifier"]
+    primary_key = "id"
+    sort_key = "batchId"
+
+    table = resource("dynamodb").Table(os.environ["DYNAMO_TABLE_NAME"])
+    item_with_id = get_item_with_id(table, primary_key, primary_key_value)
```

Comment:
There may be multiple items in the table with this id.

For example, if an asset was ingested was ingested to the Preservation System, reconciled, and removed from the lock table. Then, before the asset was downloaded to CC, another message for the same asset is picked up again by the ingest process.

Reply:
Ah, I see. But then which of these items should I add `ingested_CC =true` to; all of them?

Reply:
Yes please. We'll still want to report both ingests back to the wider department so we've put some intra-batch logic in the dr2-ingest-files-change-handler Lambda.